src/reformat.c

```diff
@@ -313,6 +333,21 @@ avifResult avifImageRGBToYUV(avifImage * image, const avifRGBImage * rgb)
                             yuvBlock[bI][bJ].y = 0.5f * rgbPixel[1] + 0.25f * (rgbPixel[0] + rgbPixel[2]);
                             yuvBlock[bI][bJ].u = 0.5f * rgbPixel[1] - 0.25f * (rgbPixel[0] + rgbPixel[2]);
                             yuvBlock[bI][bJ].v = 0.5f * (rgbPixel[0] - rgbPixel[2]);
+#if defined(AVIF_ENABLE_EXPERIMENTAL_YCGCO_R)
+                        } else if (state.mode == AVIF_REFORMAT_MODE_YCGCO_RE || state.mode == AVIF_REFORMAT_MODE_YCGCO_RO) {
+                            // Formulas from JVET-U0093.
+                            const int bitOffset = (state.mode == AVIF_REFORMAT_MODE_YCGCO_RE) ? 2 : 1;
+                            const int maxValue = (1 << (state.yuvDepth - bitOffset)) - 1;
+                            const int R = AVIF_CLAMP(avifRoundf(rgbPixel[0] * rgbMaxChannelF), 0, maxValue);
+                            const int G = AVIF_CLAMP(avifRoundf(rgbPixel[1] * rgbMaxChannelF), 0, maxValue);
+                            const int B = AVIF_CLAMP(avifRoundf(rgbPixel[2] * rgbMaxChannelF), 0, maxValue);
```

Comment:
Optional: In ISO/IEC DIS 23091-2:2022(E), the Clip3() operation is applied before the Round() operation, so the nesting of `AVIF_CLAMP` and `avifRoundf` should be reversed. For example,

```
                            assert((float)maxValue == rgbMaxChannelF);
                            const int R = avifRoundf(AVIF_CLAMP(rgbPixel[0] * rgbMaxChannelF, 0.0f, rgbMaxChannelF));
                            const int G = avifRoundf(AVIF_CLAMP(rgbPixel[1] * rgbMaxChannelF, 0.0f, rgbMaxChannelF));
                            const int B = avifRoundf(AVIF_CLAMP(rgbPixel[2] * rgbMaxChannelF, 0.0f, rgbMaxChannelF));
```

Note 1: I believe the order of Clip3() and Round() doesn't matter here. If you agree, then we don't really need to change this code. But it would be better to follow the spec more closely if it is convenient.

Note 2: JVET-U0093 applies the Clip1Y() and Round() operations in the same order. Clip1Y() is first applied in the three equations at the very end of Section 2. Round() is then applied in the first three equations at the beginning of Section 2.

Reply:
Done